src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -2,581 +2,428 @@
-        }
 
-        private static string GetModulesDescription(ProcessModuleCollection modules)
+        [Fact]
+        public void Process_GetHandle()
         {
-            var text = new StringBuilder();
-            text.AppendLine("Modules Collection:");
-            foreach (ProcessModule module in modules)
-            {
-                text.Append("    ");
-                text.AppendLine(GetModuleDescription(module));
-            }
-            return text.ToString();
+            Assert.Equal(process.Id, Interop.GetProcessId(process.SafeHandle));
         }
 
         [Fact]
-        public static void Process_MaxWorkingSet()
+        public void Process_HasExited()
         {
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
             {
-                Assert.NotNull(p.MaxWorkingSet);
+                Process p = CreateProcess();
+                p.Start();
+                p.WaitForExit();
+                Assert.True(p.HasExited, "Process_HasExited001 failed");
             }
-            finally
+
             {
-                p.Kill();
-                p.WaitForExit();
+                Process p = CreateProcessInfinite();
+                p.Start();
+                try
+                {
+                    Assert.False(p.HasExited, "Process_HasExited002 failed");
+                }
+                finally
+                {
+                    p.Kill();
+                    p.WaitForExit();
+                }
+
+                Assert.True(p.HasExited, "Process_HasExited003 failed");
             }
         }
 
         [Fact]
-        public static void Process_MinWorkingSet()
+        public void Process_MachineName()
         {
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
-            {
-                Assert.NotNull(p.MinWorkingSet);
-            }
-            finally
-            {
-                p.Kill();
-                p.WaitForExit();
-            }
+            // Checking that the MachineName returns some value.
+            Assert.NotNull(process.MachineName);
         }
 
         [Fact]
-        public static void Process_Modules()
+        public void Process_MainModule()
         {
-            // Checking that the MachineName returns some value.
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
+            // Get MainModule property from a Process object
+            string moduleName = process.MainModule.ModuleName;
+            Assert.Equal(_processName, moduleName);
+
+            // Check that the mainModule is present in the modules list.
+            bool foundMainModule = false;
+            foreach (ProcessModule pModule in process.Modules)
             {
-                foreach (ProcessModule pModule in p.Modules)
+                if (String.Equals(moduleName, _processName, StringComparison.OrdinalIgnoreCase))
```

Comment:
I agree that LINQ expressions make maintainability easier however it can be a challenge to debug and can result in multi line anonymous methods. 

I looked at the foreach() patterns in process and I agree that some of them can be changed. I will update them. Let me know if you feel strongly about replacing all the calls to LINQ instead.